js/app/historyStore.js

```diff
@@ -369,6 +375,32 @@ var HistoryStore = new Lang.Class({
         }
         this.set_current_item(HistoryItem.HistoryItem.new_from_object(target_item));
     },
+
+    can_share_current_item: function () {
+        let item = this.get_current_item();
+        return (item.model &&
+                item.model instanceof Eknc.ContentObjectModel &&
+                item.model.original_uri !== null &&
+                item.model.original_uri !== "");
+    },
+
+    social_share: function (network) {
+        if (!this.can_share_current_item())
+          return;
+        let item = this.get_current_item();
+
+	let uri = encodeURIComponent (item.model.original_uri);
+	let msg = encodeURIComponent (_("Shared from Endless"));
+
+        if (network === 'facebook')
+	  Gtk.show_uri (null, 'https://www.facebook.com/dialog/share?app_id=407909575958642&display=popup' +
+                        '&href=' + uri + '&quote=' + msg, Gdk.CURRENT_TIME);
+        else if (network === 'twitter')
+	  Gtk.show_uri (null, 'https://twitter.com/intent/tweet?hashtags=SharedFromEndless' +
+	                '&original_referer=https%3A%2F%2Fendlessos.com&url=' + uri, Gdk.CURRENT_TIME);
+        else if (network === 'whatsapp')
+	  Gtk.show_uri (null, 'https://api.whatsapp.com/send?text=' + uri + '%20' + msg, Gdk.CURRENT_TIME);
+    },
```

Comment:
maybe adding a basic unit test, to make sure this doesn't break?

Reply:
testing the actual codepath would be too complicated, but I can add a simple test that would check the url we are using to share are working